### ISSUE TITLE
error: value of type 'NSCondition' has no member 'withLock' when compiling latest git mpv on mac

### ISSUE THREAD

**Issue by eko5624** [mpv-player/mpv#13763]
Log:
```
The Meson build system
Version: 1.4.0
Source dir: /Users/runner/work/mpv-macos-intel/mpv-macos-intel/packages/mpv
Build dir: /Users/runner/work/mpv-macos-intel/mpv-macos-intel/packages/mpv/build
Build type: native build
Project name: mpv
Project version: 0.37.0-UNKNOWN
C compiler for the host machine: clang (clang 10.0.0 "Apple clang version 10.0.0 (git@github.com:apple/llvm-project.git 439a44695a0534a26557cc34b4fe650dccb46aed)")
C linker for the host machine: clang ld64 820.1
Host machine cpu family: x86_64
Host machine cpu: x86_64
Program python3 found: YES (/Users/runner/work/mpv-macos-intel/mpv-macos-intel/workspace/bin/python3)
Found pkg-config: YES (/Users/runner/work/mpv-macos-intel/mpv-macos-intel/workspace/bin/pkg-config) 0.29.2
Run-time dependency libavcodec found: YES 61.2.100
Run-time dependency libavfilter found: YES 10.0.100
Run-time dependency libavformat found: YES 61.0.100
Run-time dependency libavutil found: YES 59.6.100
Run-time dependency libswresample found: YES 5.0.100
Run-time dependency libswscale found: YES 8.0.100
Run-time dependency libplacebo found: YES 7.349.0
Run-time dependency libass found: YES 0.17.1
Compiler for C supports arguments -Werror=implicit-function-declaration: YES 
Compiler for C supports arguments -Wempty-body: YES 
Compiler for C supports arguments -Wdisabled-optimization: YES 
Compiler for C supports arguments -Wstrict-prototypes: YES 
Compiler for C supports arguments -Wno-format-zero-length: YES 
Compiler for C supports arguments -Wno-redundant-decls: YES 
Compiler for C supports arguments -Wvla: YES 
Compiler for C supports arguments -Wno-format-truncation: NO 
Compiler for C supports arguments -Wimplicit-fallthrough: YES 
Compiler for C supports arguments -fno-math-errno: YES 
Compiler for C supports arguments -Wformat -Werror=format-security: YES 
Compiler for C supports link arguments -Wl,-z,noexecstack: NO 
Compiler for C supports link arguments -Wl,--nxcompat,--no-seh,--dynamicbase: NO 
Run-time dependency dl found: YES
Library atomic found: NO
Compiler for C supports link arguments -rdynamic: YES 
Run-time dependency threads found: YES
Header "pthread.h" has symbol "pthread_condattr_setclock" with dependency threads: NO 
Run-time dependency appleframeworks found: YES (Cocoa, IOKit, QuartzCore)
Checking for function "fork" : YES 
Checking for function "ppoll" : NO 
Library windowsapp skipped: feature uwp disabled
Checking for function "glob" : YES 
Header "sys/vt.h" has symbol "VT_GETMODE" : NO 
Header "sys/consio.h" has symbol "VT_GETMODE" : NO 
Checking for function "fstatfs" : YES 
Checking for function "fstatfs" : NO 
Checking if "vector check" compiles: YES 
Program /Users/runner/work/mpv-macos-intel/mpv-macos-intel/packages/mpv/TOOLS/docutils-wrapper.py found: YES (/Users/runner/work/mpv-macos-intel/mpv-macos-intel/packages/mpv/TOOLS/docutils-wrapper.py)
Program /Users/runner/work/mpv-macos-intel/mpv-macos-intel/packages/mpv/TOOLS/file2string.py found: YES (/Users/runner/work/mpv-macos-intel/mpv-macos-intel/packages/mpv/TOOLS/file2string.py)
Program /Users/runner/work/mpv-macos-intel/mpv-macos-intel/packages/mpv/TOOLS/matroska.py found: YES (/Users/runner/work/mpv-macos-intel/mpv-macos-intel/packages/mpv/TOOLS/matroska.py)
Dependency libcdio skipped: feature cdda disabled
Dependency libcdio_paranoia skipped: feature cdda disabled
Has header "linux/dvb/frontend.h" : NO 
Dependency dvdnav skipped: feature dvdnav disabled
Dependency dvdread skipped: feature dvdnav disabled
Dependency iconv skipped: feature iconv disabled
Run-time dependency mujs found: YES 1.3.3
Run-time dependency lcms2 found: YES 2.16
Found CMake: /Users/runner/work/mpv-macos-intel/mpv-macos-intel/workspace/bin/cmake (3.28.1)
Run-time dependency libarchive found: NO (tried pkgconfig, framework and cmake)
Run-time dependency libavdevice found: YES 61.0.100
Run-time dependency libbluray found: YES 1.3.4
Library m found: YES
Library rt found: NO
Run-time dependency lua found: NO (tried pkgconfig, framework and cmake)
Run-time dependency lua52 found: NO (tried pkgconfig, framework and cmake)
Run-time dependency lua5.2 found: NO (tried pkgconfig, framework and cmake)
Run-time dependency lua-5.2 found: NO (tried pkgconfig, framework and cmake)
Run-time dependency luajit found: YES 2.1.1710[398](https://github.com/eko5624/mpv-macos-intel/actions/runs/8409582546/job/23027043461#step:8:399)010
Run-time dependency rubberband found: YES 3.3.0
Dependency sdl2 skipped: feature sdl2 disabled
Dependency uchardet skipped: feature uchardet disabled
Run-time dependency vapoursynth found: NO (tried pkgconfig, framework and cmake)
Run-time dependency vapoursynth-script found: NO (tried pkgconfig, framework and cmake)
Run-time dependency zimg found: YES 3.0.5
Run-time dependency zlib found: YES 1.3
Run-time dependency alsa found: NO (tried pkgconfig, framework and cmake)
Run-time dependency appleframeworks found: YES (Foundation, AudioToolbox)
Header "AudioToolbox/AudioToolbox.h" has symbol "kAudioUnitSubType_RemoteIO" : NO 
Run-time dependency appleframeworks found: YES (CoreFoundation, CoreAudio, AudioUnit, AudioToolbox)
Run-time dependency jack found: NO (tried pkgconfig, framework and cmake)
Dependency openal skipped: feature openal disabled
Library OpenSLES found: NO
Header "sys/soundcard.h" has symbol "SNDCTL_DSP_HALT" : NO 
Run-time dependency libpipewire-0.3 found: NO (tried pkgconfig, framework and cmake)
Run-time dependency libpulse found: NO (tried pkgconfig, framework and cmake)
Run-time dependency sndio found: NO (tried pkgconfig, framework and cmake)
Header "audioclient.h" has symbol "IAudioClient" : NO 
Run-time dependency caca found: YES 0.99.beta20
Has header "d3d9.h" skipped: feature direct3d disabled
Run-time dependency libdrm found: NO (tried pkgconfig, framework and cmake)
Run-time dependency gbm found: NO (tried pkgconfig, framework and cmake)
Run-time dependency libjpeg found: YES 3.0.3
Dependency shaderc skipped: feature shaderc disabled
Run-time dependency libsixel found: NO (tried pkgconfig, framework and cmake)
Checking for function "shm_open" : YES 
Dependency spirv-cross-c-shared skipped: feature spirv-cross disabled
Run-time dependency wayland-client found: NO (tried pkgconfig, framework and cmake)
Run-time dependency wayland-cursor found: NO (tried pkgconfig, framework and cmake)
Run-time dependency wayland-protocols found: NO (tried pkgconfig, framework and cmake)
Run-time dependency xkbcommon found: NO (tried pkgconfig, framework and cmake)
Run-time dependency wayland-client found: NO (tried pkgconfig, framework and cmake)
Has header "linux/input-event-codes.h" with dependency wayland-client: NO 
Program wayland-scanner found: NO
Dependency x11 found: NO unknown version, but need: ['>= 1.0.0']
Run-time dependency x11 found: NO (tried pkgconfig, framework and cmake)
Run-time dependency xscrnsaver found: NO (tried pkgconfig, framework and cmake)
Run-time dependency xext found: NO (tried pkgconfig, framework and cmake)
Run-time dependency xpresent found: NO (tried pkgconfig, framework and cmake)
Run-time dependency xrandr found: NO (tried pkgconfig, framework and cmake)
Dependency xv skipped: feature xv disabled
Run-time dependency appleframeworks found: YES (OpenGL)
Library EGL found: NO
Run-time dependency egl found: NO (tried pkgconfig, framework and cmake)
Library EGL skipped: feature egl-android disabled
Run-time dependency wayland-egl found: NO (tried pkgconfig, framework and cmake)
Run-time dependency vulkan found: YES 1.3.281
Checking for function "vkCreateDisplayPlaneSurfaceKHR" with dependency vulkan: NO 
Run-time dependency ffnvcodec found: NO (tried pkgconfig, framework and cmake)
Header "OpenGLES/ES3/glext.h" has symbol "GL_RGB32F" : NO 
Run-time dependency libva found: NO (tried pkgconfig, framework and cmake)
Dependency libva-drm skipped: feature vaapi-drm disabled
Dependency libva-wayland skipped: feature vaapi-wayland disabled
Dependency libva-x11 skipped: feature vaapi-x11 disabled
Dependency libva-win32 skipped: feature vaapi-win32 disabled
Dependency vdpau skipped: feature vdpau disabled
Run-time dependency appleframeworks found: YES (CoreVideo)
Program /Users/runner/work/mpv-macos-intel/mpv-macos-intel/packages/mpv/TOOLS/macos-sdk-version.py found: YES (/Users/runner/work/mpv-macos-intel/mpv-macos-intel/packages/mpv/TOOLS/macos-sdk-version.py)
Message: Detected macOS sdk path: /Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk
Message: Detected macOS SDK: 11.3
meson.build:1485: WARNING: add_languages is missing native:, assuming languages are wanted for both host and build.
Objective-C compiler for the host machine: cc (clang 10.0.0)
Objective-C linker for the host machine: cc ld64 820.1
Program xcrun found: YES (/usr/bin/xcrun)
Program /Library/Developer/Toolchains/swift-5.4-RELEASE.xctoolchain/usr/bin/swift found: YES (/Library/Developer/Toolchains/swift-5.4-RELEASE.xctoolchain/usr/bin/swift)
Message: Detected Swift version: 5.4
Has header "AppKit/NSTouchBar.h" : YES 
Program /Users/runner/work/mpv-macos-intel/mpv-macos-intel/packages/mpv/TOOLS/macos-swift-lib-directory.py found: YES (/Users/runner/work/mpv-macos-intel/mpv-macos-intel/packages/mpv/TOOLS/macos-swift-lib-directory.py)
Message: Detected Swift library directory: /Library/Developer/Toolchains/swift-5.4-RELEASE.xctoolchain/usr/lib/swift/macosx
Program rst2man rst2man.py skipped: feature manpage-build disabled
Program rst2html rst2html.py skipped: feature html-build disabled
Program rst2pdf skipped: feature pdf-build disabled
Configuring config.h using configuration
Message: List of enabled features: av-channel-layout avif-muxer bsd-fstatfs build-date caca cocoa coreaudio cplugins darwin ffmpeg gl gl-cocoa glob glob-posix gpl javascript jpeg jpegxl lavu-uuid lcms2 libass libavdevice libbluray libdl libm libplacebo luajit mac-thread-name macos-cocoa-cb macos-media-player macos-touchbar posix posix-shm rubberband rubberband-3 swift threads vector videotoolbox-gl videotoolbox-pl vulkan vulkan-interop zimg zimg-st[428](https://github.com/eko5624/mpv-macos-intel/actions/runs/8409582546/job/23027043461#step:8:429) zlib
Fetching value of define "MPV_CLIENT_API_VERSION" : (((2) << 16) | (3) | 0UL) 
Compiler for C supports link arguments -Wl,-Bsymbolic: NO 
Program /Users/runner/work/mpv-macos-intel/mpv-macos-intel/packages/mpv/TOOLS/osxbundle.py found: YES (/Users/runner/work/mpv-macos-intel/mpv-macos-intel/packages/mpv/TOOLS/osxbundle.py)
Build targets in project: 25

mpv 0.37.0-UNKNOWN

    d3d11        : NO
    javascript   : YES
    libmpv       : NO
    lua          : YES
    opengl       : YES
    vulkan       : YES
    wayland      : NO
    x11          : NO

  User defined options
    buildtype    : release
    libdir       : /Users/runner/work/mpv-macos-intel/mpv-macos-intel/workspace/lib
    prefix       : /Users/runner/work/mpv-macos-intel/mpv-macos-intel/workspace
    iconv        : disabled
    manpage-build: disabled
    swift-flags  : -target x86_64-apple-macos11.0

Found ninja-1.11.1.git.kitware.jobserver-1 at /Users/runner/work/mpv-macos-intel/mpv-macos-intel/workspace/bin/ninja
ninja: Entering directory `/Users/runner/work/mpv-macos-intel/mpv-macos-intel/packages/mpv/build'
[1/241] Generating etc/mpv-icon-8bit-16x16.png with a custom command
sys:1: ResourceWarning: unclosed file <_io.TextIOWrapper name='etc/mpv-icon-8bit-16x16.png.inc' mode='w' encoding='UTF-8'>
[2/241] Generating etc/mpv-icon-8bit-64x64.png with a custom command
sys:1: ResourceWarning: unclosed file <_io.TextIOWrapper name='etc/mpv-icon-8bit-64x64.png.inc' mode='w' encoding='UTF-8'>
[3/241] Generating etc/mpv-icon-8bit-32x32.png with a custom command
sys:1: ResourceWarning: unclosed file <_io.TextIOWrapper name='etc/mpv-icon-8bit-32x32.png.inc' mode='w' encoding='UTF-8'>
[4/241] Generating etc/input.conf with a custom command
sys:1: ResourceWarning: unclosed file <_io.TextIOWrapper name='etc/input.conf.inc' mode='w' encoding='UTF-8'>
[5/241] Generating etc/mpv-icon-8bit-128x128.png with a custom command
sys:1: ResourceWarning: unclosed file <_io.TextIOWrapper name='etc/mpv-icon-8bit-128x128.png.inc' mode='w' encoding='UTF-8'>
[6/241] Generating ebml_defs.inc with a custom command
/Users/runner/work/mpv-macos-intel/mpv-macos-intel/packages/mpv/TOOLS/matroska.py:469: ResourceWarning: unclosed file <_io.TextIOWrapper name='ebml_defs.inc' mode='w' encoding='UTF-8'>
  generate_C_definitions(open(sys.argv[2], "w"))
ResourceWarning: Enable tracemalloc to get the object allocation traceback
[7/241] Generating ebml_types.h with a custom command
/Users/runner/work/mpv-macos-intel/mpv-macos-intel/packages/mpv/TOOLS/matroska.py:[467](https://github.com/eko5624/mpv-macos-intel/actions/runs/8409582546/job/23027043461#step:8:468): ResourceWarning: unclosed file <_io.TextIOWrapper name='ebml_types.h' mode='w' encoding='UTF-8'>
  generate_C_header(open(sys.argv[2], "w"))
ResourceWarning: Enable tracemalloc to get the object allocation traceback
[8/241] Generating etc/builtin.conf with a custom command
sys:1: ResourceWarning: unclosed file <_io.TextIOWrapper name='etc/builtin.conf.inc' mode='w' encoding='UTF-8'>
[9/241] Generating player/lua/defaults.lua with a custom command
sys:1: ResourceWarning: unclosed file <_io.TextIOWrapper name='player/lua/defaults.lua.inc' mode='w' encoding='UTF-8'>
[10/241] Generating player/lua/assdraw.lua with a custom command
sys:1: ResourceWarning: unclosed file <_io.TextIOWrapper name='player/lua/assdraw.lua.inc' mode='w' encoding='UTF-8'>
[11/241] Generating player/lua/options.lua with a custom command
sys:1: ResourceWarning: unclosed file <_io.TextIOWrapper name='player/lua/options.lua.inc' mode='w' encoding='UTF-8'>
[12/241] Generating player/javascript/defaults.js with a custom command
sys:1: ResourceWarning: unclosed file <_io.TextIOWrapper name='player/javascript/defaults.js.inc' mode='w' encoding='UTF-8'>
[13/241] Generating player/lua/ytdl_hook.lua with a custom command
sys:1: ResourceWarning: unclosed file <_io.TextIOWrapper name='player/lua/ytdl_hook.lua.inc' mode='w' encoding='UTF-8'>
[14/241] Generating player/lua/stats.lua with a custom command
sys:1: ResourceWarning: unclosed file <_io.TextIOWrapper name='player/lua/stats.lua.inc' mode='w' encoding='UTF-8'>
[15/241] Generating player/lua/auto_profiles.lua with a custom command
sys:1: ResourceWarning: unclosed file <_io.TextIOWrapper name='player/lua/auto_profiles.lua.inc' mode='w' encoding='UTF-8'>
[16/241] Generating player/lua/console.lua with a custom command
sys:1: ResourceWarning: unclosed file <_io.TextIOWrapper name='player/lua/console.lua.inc' mode='w' encoding='UTF-8'>
[17/241] Generating player/lua/input.lua with a custom command
sys:1: ResourceWarning: unclosed file <_io.TextIOWrapper name='player/lua/input.lua.inc' mode='w' encoding='UTF-8'>
[18/241] Generating sub/osd_font.otf with a custom command
sys:1: ResourceWarning: unclosed file <_io.TextIOWrapper name='sub/osd_font.otf.inc' mode='w' encoding='UTF-8'>
[19/241] Generating player/lua/osc.lua with a custom command
sys:1: ResourceWarning: unclosed file <_io.TextIOWrapper name='player/lua/osc.lua.inc' mode='w' encoding='UTF-8'>
[20/241] Generating common/version.h with a custom command
[21/241] Compiling C object mpv.p/osdep_main-fn-mac.c.o
[22/241] Generating TOOLS/osxbundle/osxbundle with a custom command
sys:1: ResourceWarning: unclosed file <_io.TextIOWrapper name='TOOLS/osxbundle/icon.icns.inc' mode='w' encoding='UTF-8'>
[23/241] Generating osdep/mac/swift_targets with a custom command
FAILED: osdep/mac/swift.swiftmodule osdep/mac/swift.h osdep/mac/swift.o 
/Library/Developer/Toolchains/swift-5.4-RELEASE.xctoolchain/usr/bin/swift -frontend -c -sdk /Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk -enable-objc-interop -emit-objc-header -parse-as-library -O -D HAVE_MACOS_TOUCHBAR -target x86_64-apple-macos11.0 -module-name swift -emit-module-path osdep/mac/swift.swiftmodule -import-objc-header /Users/runner/work/mpv-macos-intel/mpv-macos-intel/packages/mpv/osdep/mac/swift_bridge.h -emit-objc-header-path osdep/mac/swift.h -o osdep/mac/swift.o ../osdep/mac/libmpv_helper.swift ../osdep/mac/input_helper.swift ../osdep/mac/log_helper.swift ../osdep/mac/mpv_helper.swift ../osdep/mac/menu_bar.swift ../osdep/mac/precise_timer.swift ../osdep/mac/swift_compat.swift ../osdep/mac/swift_extensions.swift ../video/out/mac/common.swift ../video/out/mac/title_bar.swift ../video/out/mac/view.swift ../video/out/mac/window.swift ../video/out/cocoa_cb_common.swift ../video/out/mac/gl_layer.swift ../video/out/mac_common.swift ../video/out/mac/metal_layer.swift ../osdep/mac/remote_command_center.swift -I. -I/Users/runner/work/mpv-macos-intel/mpv-macos-intel/packages/mpv -I/Users/runner/work/mpv-macos-intel/mpv-macos-intel/workspace/include
../osdep/mac/libmpv_helper.swift:150:20: warning: 'CGLFlushDrawable' was deprecated in macOS 10.14: OpenGL API deprecated. (Define GL_SILENCE_DEPRECATION to silence these warnings)
        if !skip { CGLFlushDrawable(ctx) }
                   ^
../osdep/mac/input_helper.swift:76:14: error: value of type 'NSCondition' has no member 'withLock'
        lock.withLock {
        ~~~~ ^~~~~~~~
../osdep/mac/input_helper.swift:101:21: error: value of type 'NSCondition' has no member 'withLock'
        return lock.withLock {
               ~~~~ ^~~~~~~~
../osdep/mac/input_helper.swift:101:30: error: unable to infer complex closure return type; add explicit type to disambiguate
        return lock.withLock {
                             ^
                               () -> <#Result#> in 
../osdep/mac/input_helper.swift:119:14: error: value of type 'NSCondition' has no member 'withLock'
        lock.withLock {
        ~~~~ ^~~~~~~~
../osdep/mac/input_helper.swift:119:23: error: unable to infer complex closure return type; add explicit type to disambiguate
        lock.withLock {
                      ^
                        () -> <#Result#> in 
../osdep/mac/input_helper.swift:126:14: error: value of type 'NSCondition' has no member 'withLock'
        lock.withLock {
        ~~~~ ^~~~~~~~
../osdep/mac/input_helper.swift:126:23: error: unable to infer complex closure return type; add explicit type to disambiguate
        lock.withLock {
                      ^
                        () -> <#Result#> in 
../osdep/mac/input_helper.swift:133:14: error: value of type 'NSCondition' has no member 'withLock'
        lock.withLock {
        ~~~~ ^~~~~~~~
../osdep/mac/input_helper.swift:140:14: error: value of type 'NSCondition' has no member 'withLock'
        lock.withLock {
        ~~~~ ^~~~~~~~
../osdep/mac/input_helper.swift:147:14: error: value of type 'NSCondition' has no member 'withLock'
        lock.withLock {
        ~~~~ ^~~~~~~~
../osdep/mac/input_helper.swift:147:23: error: unable to infer complex closure return type; add explicit type to disambiguate
        lock.withLock {
                      ^
                        () -> <#Result#> in 
../osdep/mac/input_helper.swift:186:14: error: value of type 'NSCondition' has no member 'withLock'
        lock.withLock {
        ~~~~ ^~~~~~~~
../osdep/mac/input_helper.swift:208:14: error: value of type 'NSCondition' has no member 'withLock'
        lock.withLock {
        ~~~~ ^~~~~~~~
../osdep/mac/input_helper.swift:215:14: error: value of type 'NSCondition' has no member 'withLock'
        lock.withLock {
        ~~~~ ^~~~~~~~
../osdep/mac/input_helper.swift:222:14: error: value of type 'NSCondition' has no member 'withLock'
        lock.withLock { while input == nil { lock.wait() } }
        ~~~~ ^~~~~~~~
../video/out/mac/title_bar.swift:39:29: warning: 'light' was deprecated in macOS 10.14: Use a semantic material instead.  To force the appearance of a view hierarchy, set the `appearance` property to an appropriate NSAppearance value.
            if material == .light || material == .dark || material == .mediumLight ||
                            ^
../video/out/mac/title_bar.swift:39:51: warning: 'dark' was deprecated in macOS 10.14: Use a semantic material instead.  To force the appearance of a view hierarchy, set the `appearance` property to an appropriate NSAppearance value.
            if material == .light || material == .dark || material == .mediumLight ||
                                                  ^
../video/out/mac/title_bar.swift:39:72: warning: 'mediumLight' was deprecated in macOS 10.14: Use a semantic material instead.  To force the appearance of a view hierarchy, set the `appearance` property to an appropriate NSAppearance value.
            if material == .light || material == .dark || material == .mediumLight ||
                                                                       ^
../video/out/mac/title_bar.swift:40:29: warning: 'ultraDark' was deprecated in macOS 10.14: Use a semantic material instead.  To force the appearance of a view hierarchy, set the `appearance` property to an appropriate NSAppearance value.
               material == .ultraDark
                            ^
../video/out/mac/title_bar.swift:216:53: warning: 'dark' was deprecated in macOS 10.14: Use a semantic material instead.  To force the appearance of a view hierarchy, set the `appearance` property to an appropriate NSAppearance value.
        case "14", "dark":                  return .dark
                                                    ^
../video/out/mac/title_bar.swift:217:53: warning: 'light' was deprecated in macOS 10.14: Use a semantic material instead.  To force the appearance of a view hierarchy, set the `appearance` property to an appropriate NSAppearance value.
        case "15", "light":                 return .light
                                                    ^
../video/out/mac/title_bar.swift:218:53: warning: 'mediumLight' was deprecated in macOS 10.14: Use a semantic material instead.  To force the appearance of a view hierarchy, set the `appearance` property to an appropriate NSAppearance value.
        case "16", "mediumLight":           return .mediumLight
                                                    ^
../video/out/mac/title_bar.swift:219:53: warning: 'ultraDark' was deprecated in macOS 10.14: Use a semantic material instead.  To force the appearance of a view hierarchy, set the `appearance` property to an appropriate NSAppearance value.
        case "17", "ultraDark":             return .ultraDark
                                                    ^
../video/out/mac/view.swift:36:9: warning: 'wantsBestResolutionOpenGLSurface' was deprecated in macOS 10.14: Use NSOpenGLView instead.
        wantsBestResolutionOpenGLSurface = true
        ^
../video/out/mac/gl_layer.swift:70:16: warning: 'CAOpenGLLayer' was deprecated in macOS 10.14: OpenGL is deprecated
class GLLayer: CAOpenGLLayer {
               ^
../video/out/mac/gl_layer.swift:115:9: warning: 'CGLSetParameter' was deprecated in macOS 10.14: OpenGL API deprecated. (Define GL_SILENCE_DEPRECATION to silence these warnings)
        CGLSetParameter(cglContext, kCGLCPSwapInterval, &i)
        ^
../video/out/mac/gl_layer.swift:188:13: warning: 'NSDisableScreenUpdates()' was deprecated in macOS 10.11: As of 10.11 it is not generally necessary to take explicit action to achieve visual atomicity. +[NSAnimationContext runAnimationGroup:] and other similar methods can be used when a stronger than normal need for visual atomicity is required. The NSAnimationContext methods do not suffer from the same performance problems as NSDisableScreenUpdates.
            NSDisableScreenUpdates()
            ^
../video/out/mac/gl_layer.swift:195:13: warning: 'NSEnableScreenUpdates()' was deprecated in macOS 10.11: As of 10.11 it is not generally necessary to take explicit action to achieve visual atomicity. +[NSAnimationContext runAnimationGroup:] and other similar methods can be used when a stronger than normal need for visual atomicity is required. The NSAnimationContext methods do not suffer from the same performance problems as NSEnableScreenUpdates.
            NSEnableScreenUpdates()
            ^
../video/out/mac/gl_layer.swift:291:23: warning: 'CGLChoosePixelFormat' was deprecated in macOS 10.14: OpenGL API deprecated. (Define GL_SILENCE_DEPRECATION to silence these warnings)
                err = CGLChoosePixelFormat(format, &pix, &npix)
                      ^
../video/out/mac/gl_layer.swift:307:36: warning: 'CGLErrorString' was deprecated in macOS 10.14: OpenGL API deprecated. (Define GL_SILENCE_DEPRECATION to silence these warnings)
        let errS = String(cString: CGLErrorString(err))
                                   ^
../video/out/mac/gl_layer.swift:320:21: warning: 'CGLCreateContext' was deprecated in macOS 10.14: OpenGL API deprecated. (Define GL_SILENCE_DEPRECATION to silence these warnings)
        let error = CGLCreateContext(pixelFormat, nil, &context)
                    ^
../video/out/mac/gl_layer.swift:323:40: warning: 'CGLErrorString' was deprecated in macOS 10.14: OpenGL API deprecated. (Define GL_SILENCE_DEPRECATION to silence these warnings)
            let errS = String(cString: CGLErrorString(error))
                                       ^
ninja: build stopped: subcommand failed.
INFO: autodetecting backend as ninja
INFO: calculating backend command to run: /Users/runner/work/mpv-macos-intel/mpv-macos-intel/workspace/bin/ninja -C /Users/runner/work/mpv-macos-intel/mpv-macos-intel/packages/mpv/build
Error: Process completed with exit code 1.
```

**Comment by Akemi** [comment]
at least take so much time to properly fill out the issue template.

your setup seems botched:
```
Message: Detected macOS sdk path: /Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk
Message: Detected macOS SDK: 11.3
...
Program /Library/Developer/Toolchains/swift-5.4-RELEASE.xctoolchain/usr/bin/swift found: YES (/Library/Developer/Toolchains/swift-5.4-RELEASE.xctoolchain/usr/bin/swift)
Message: Detected Swift version: 5.4
```

xcode 14.2 contains SDK 13.1 not 11.3. also the proper path for swift should be within xcode (/Applications/Xcode_14.2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/swift) and not the command line tools. that would also be swift version 5.7.2 and not 5.4.

[withLock](https://developer.apple.com/documentation/foundation/nslocking/4059821-withlock) ([NSCondition](https://developer.apple.com/documentation/foundation/nscondition) conforms to NSLocking) was added with xcode 14. xcode 14 contained at least swift 5.7. sadly apple doesn't clearly mention which swift versions added support for specific feature and it can probably only be inferred by the minimum xcode version.

TLDR your swift is probably too old and your setup seems also botched. we should also probably up our minimum supported swift version.

see also our own [runner with xdoce 14.2](https://github.com/mpv-player/mpv/actions/runs/8409621524/job/23027116800?pr=13758).

**Comment by eko5624** [comment]
I delete SDK 13.1 in xcode 14.2 and add SDK 11.3 for the backward compatible of macos-11.
Below is a part of workflow action:
```
name: build

on:
  #schedule:
  #  - cron: '10 0 * * MON'
    
  workflow_dispatch:   

jobs:
  build:
    runs-on: macos-13
    env:
      CC: clang
      CXX: clang++
      MACOSX_DEPLOYMENT_TARGET: 11
      SDKROOT: "/Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk"
      MACOS_SDK: "/Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk"
      SDK_PATH: "/Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs"
      EXTRA_CFLAGS: "-mmacosx-version-min=11 -isysroot /Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk"
      EXTRA_CXXFLAGS: "-mmacosx-version-min=11 -isysroot /Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk"
      EXTRA_LDFLAGS: "-mmacosx-version-min=11 -isysroot /Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk"
      SWIFT_FLAGS: "-target x86_64-apple-macosx11.0"
      CURL_RETRIES: "--connect-timeout 60 --retry 5 --retry-delay 5"
    steps:    
    - name: Checkout
      uses: actions/checkout@main

    - name: Switch to MacOSX 11 SDK
      run: |
        #curl -OL https://github.com/phracker/MacOSX-SDKs/releases/download/11.3/MacOSX11.3.sdk.tar.xz
        tar -C $SDK_PATH -xf MacOSX11.3.sdk.tar.xz
        sudo rm -rf $SDK_PATH/MacOSX13.1.sdk
        sudo rm -rf $SDK_PATH/MacOSX.sdk
        sudo ln -s $SDK_PATH/MacOSX11.3.sdk $SDK_PATH/MacOSX.sdk
        sudo /usr/libexec/PlistBuddy -c "Set :MinimumSDKVersion 11.3" /Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Info.plist

    - name: Install Swift 5.x toolchain
      run: |
        curl -ksJLO https://download.swift.org/swift-5.4-release/xcode/swift-5.4-RELEASE/swift-5.4-RELEASE-osx.pkg
        sudo installer -pkg swift-5.4-RELEASE-osx.pkg -target /

    - name: Switch to Xcode 14.2
      run: |
        sudo xcode-select --reset
        sudo xcode-select -s /Applications/Xcode_14.2.app/Contents/Developer  
 
    - name: Remove stray upstream python binary symlinks under /usr/local
      run: |
        find /usr/local/bin -lname '*/Library/Frameworks/Python.framework/*' -delete -print
        brew unlink python
```
I will try to remove `Install Swift 5.x toolchain`, and see if it works.

**Comment by Akemi** [comment]
> I delete SDK 13.1 in xcode 14.2 and add SDK 11.3 for the backward compatible of macos-11.

that is completely unnecessary. you can build with newer macOS SDKs for older versions of macOS. you just just have to properly set the target (min) macOS version for both swift and clang (env variables and/or compiler switches).

**Comment by eko5624** [comment]
Thanks for your infomation, issue resolved.

**Comment by Akemi** [comment]
just on a side note. back when i tested the [toolchains from swift.org](https://github.com/mpv-player/mpv/pull/6703) they were bundled with the macOS SDK versions it should be used with. though that could have possibly be changed.

also if you really want to override the SDK, SDK version or swift lib info you can use the environment variables [MACOS_SDK, MACOS_SDK_VERSION](https://github.com/mpv-player/mpv/blob/master/TOOLS/macos-sdk-version.py#L15-L16) and [SWIFT_LIB_DYNAMIC](https://github.com/mpv-player/mpv/blob/master/TOOLS/macos-swift-lib-directory.py#L12).
overriding the swift compiler and the swift version doesn't work anymore, this was removed with the switch of our build system to meson.
